darwin_stop_world.c

```diff
@@ -684,8 +663,10 @@ GC_INLINE void GC_thread_resume(thread_act_t thread)
 # endif
   /* Resume the thread */
   kern_result = thread_resume(thread);
+# ifdef DEBUG_THREADS
   if (kern_result != KERN_SUCCESS)
-    ABORT("thread_resume failed");
+    GC_log_printf("thread_resume failed %p\n", (void *)thread);
```

Comment:
what are the cases when we should ignore thread_resume failure? can it happen when GC_thread_resume is called having GC_query_task_threads is false?

Reply:
during having stress test on this I faced the issue when it failed to resume previously successfully suspended thread. it is rare case I met it like several times a day. Had no clear understanding why it was happening under iOS12. Error code usually was "mach port invalid"
considering that current logic allows failure to suspend thread as normal case -- it might be ok to allow application survive when there is thread resume failure as well.

Reply:
> Error code usually was "mach port invalid"
KERN_INVALID_ARGUMENT error code? Any other error code?

Reply:
1. Have you tested the path in 2 variants:  CFLAGS_EXTRA=-DGC_NO_THREADS_DISCOVERY and CFLAGS_EXTRA=-DGC_DISCOVER_TASK_THREADS.

2. What is the value of GC_query_task_threads when you meat resume failure? Is it always false, or  always true, or any?

Reply:
hi @ivmai 
I played a little with stress test today to get exact error number but was not able to reproduce on recent ios12 beta. will check later today and come back with result. 

> Have you tested the path in 2 variants: CFLAGS_EXTRA=-DGC_NO_THREADS_DISCOVERY and CFLAGS_EXTRA=-DGC_DISCOVER_TASK_THREADS.

very first "fix" was to use `--disable-threads-discovery` at my end. It is not big surprise it worked as each managed thread in this case keeps own thread's match port (e.g. no dealloc happening)

> What is the value of GC_query_task_threads when you meat resume failure? Is it always false, or always true, or any?

Sorry I have no this information as I'm stress testing only isolated stop/start world code snippets in very empty Xcode project without bdwgc itself.